From 3f64676a238535434c8438159920e4268ff26261 Mon Sep 17 00:00:00 2001
From: Josh Wu <josh.wu@atmel.com>
Date: Wed, 12 Feb 2014 18:02:46 +0800
Subject: [PATCH] arm: atmel: sama5d3x: add a new environment 'ek_name'

This 'ek_name' can be use to know the cpu type in runtime.
ek_name can be 'sama5d31ek', 'sama5d33ek', 'sama5d34ek', 'sama5d35ek'
and 'sama5d36ek'.

Signed-off-by: Josh Wu <josh.wu@atmel.com>
---
 board/atmel/sama5d3xek/sama5d3xek.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/board/atmel/sama5d3xek/sama5d3xek.c b/board/atmel/sama5d3xek/sama5d3xek.c
index ae88092..0d05789 100644
--- a/board/atmel/sama5d3xek/sama5d3xek.c
+++ b/board/atmel/sama5d3xek/sama5d3xek.c
@@ -347,12 +347,20 @@ void spi_cs_deactivate(struct spi_slave *slave)
 #endif /* CONFIG_ATMEL_SPI */
 
 #ifdef CONFIG_BOARD_LATE_INIT
+#include <linux/ctype.h>
 int board_late_init(void)
 {
 #ifdef CONFIG_ENV_VARS_UBOOT_RUNTIME_CONFIG
+	const char *SAMA5D3_BOARD_EK_NAME = "ek_name";
 	const char *SAMA5D3_MB_REV_ENV_NAME = "mb_rev";
 	const char *SAMA5D3_DM_TYPE_ENV_NAME = "dm_type";
 	char rev_code[2], dm_id;
+	char name[32], *p;
+
+	strcpy(name, get_cpu_name());
+	for (p = name; *p != '\0'; *p = tolower(*p), p++);
+	strcat(name, "ek");
+	setenv(SAMA5D3_BOARD_EK_NAME, name);
 
 	rev_code[0] = get_mb_rev_code();
 	rev_code[1] = '\0';
-- 
1.9.1

